src/main/java/net/sf/jabref/shared/DBMSProcessor.java

```diff
@@ -46,9 +45,36 @@ public DBMSProcessor(Connection connection) {
      * @throws SQLException
      */
     public boolean checkBaseIntegrity() throws SQLException {
-        List<String> requiredTables = new ArrayList<>(Arrays.asList("ENTRY", "FIELD", "METADATA")); // the list should be dynamic
-        DatabaseMetaData databaseMetaData = connection.getMetaData();
+        return checkTableAvailibility("ENTRY", "FIELD", "METADATA");
+    }
+
+    /**
+     * Determines whether the database is using an old structure.
+     * @return <code>true</code> if the structure is old, else <code>false</code>.
+     */
+    public boolean checkOldIntergrity() throws SQLException {
```

Comment:
Maybe, the whole method should be called `checkIfDatabaseIs3dot6OrNewer` (not sure whether that is really the right name)